### PR TITLE
[Release-1.2.2] ADL Upgrade to AutoRest

### PR DESCRIPTION
**Pull request by begoldsm**
This is the PowerShell change to update ADL cmdlets to AutoRest. Rebasing the PR to the release branch instead of dev.

### CODE REVIEW

**Comment by begoldsm** [comment]
PowerShell On Demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/562/

PowerShell Sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/274/

File: src/Common/Commands.Common.Authentication/Factories/ClientFactory.cs [hunk 1]
```diff
@@ -119,11 +119,18 @@ public ClientFactory()
...
-        /// <param name="endpoint"></param>
+        /// <typeparam name="TClient">The type of the client.</typeparam>
+        /// <param name="profile">The profile.</param>
+        /// <param name="subscription">The subscription.</param>
+        /// <param name="endpoint">The endpoint.</param>
         /// <returns></returns>
+        /// <exception cref="System.ApplicationException"></exception>
+        /// <exception cref="System.ArgumentException">
+        /// accountName
+        /// or
```
Comment (hovsepm) [inline src/Common/Commands.Common.Authentication/Factories/ClientFactory.cs]: or what?

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands/WaitAzureRmDataLakeAnalyticsJob.cs [hunk 2]
```diff
@@ -75,7 +70,7 @@ public override void ExecuteCmdlet()
                 WriteVerboseWithTimestamp(string.Format(Resources.WaitJobState, jobInfo.State));
                 Thread.Sleep(WaitIntervalInSeconds*1000);
```
Comment (hovsepm) [inline src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands/WaitAzureRmDataLakeAnalyticsJob.cs]: @begoldsm please change this to TestMockSupport.Delay() method call.

File: src/Common/Commands.Common.Authentication/Factories/ClientFactory.cs (hunk 1 above)
Comment (begoldsm) [inline src/Common/Commands.Common.Authentication/Factories/ClientFactory.cs]: Ah, this was a ghostDoc generation. I will fix it.

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands/WaitAzureRmDataLakeAnalyticsJob.cs (hunk 2 above)
Comment (begoldsm) [inline src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands/WaitAzureRmDataLakeAnalyticsJob.cs]: Fixed, and I found other places where I was calling Thread.Sleep directly and replaced it with this.

**Pull request merged**